servers/src/main/java/tachyon/worker/eviction/EvictLRUBase.java

```diff
@@ -41,11 +49,44 @@ protected EvictLRUBase(boolean lastTier) {
    * @return true if the block can be evicted, false otherwise
    */
   protected boolean blockEvictable(long blockId, Set<Integer> pinList) {
-    return !mLastTier || !pinList.contains(BlockInfo.computeInodeId(blockId));
+    return !mLastTier || !pinList.contains(tachyon.master.BlockInfo.computeInodeId(blockId));
+  }
+
+  /**
+   * Clean the blocks that have been evicted from evicting list in each StorageDir
+   */
+  protected void cleanEvictingBlockIds() {
+    List<Pair<StorageDir, Long>> outdatedBlocks = new ArrayList<Pair<StorageDir, Long>>();
+    for (Entry<StorageDir, Long> block : mEvictingBlockIds.entries()) {
+      StorageDir dir = block.getKey();
+      long blockId = block.getValue();
+      // If the block has been evicted, mark it outdated
+      if (!dir.containsBlock(blockId)) {
+        outdatedBlocks.add(new Pair<StorageDir, Long>(dir, blockId));
```

Comment:
Oh I see, instead of doing the for each you may be able to use an iterator and the `remove()` method to do this in one pass.